ext/preprocess/FileSystemGlobbingInternalization.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.Framework.Runtime;
+
+namespace NuGet.Compiler.Preprocess
+{
+    public class FileSystemGlobbingInternalization : ICompileModule
+    {
+        private readonly string _appBase;
+
+        public FileSystemGlobbingInternalization(IServiceProvider services)
+        {
+            var appEnv = services.GetService(typeof(IApplicationEnvironment)) as IApplicationEnvironment;
+            _appBase = appEnv.ApplicationBasePath;
+        }
+
+        public void BeforeCompile(IBeforeCompileContext context)
+        {
+            var fileSystemGlobbingDir = Path.Combine(_appBase, "..", "..", "submodules", "FileSystem", "src",
```

Comment:
We need this PR (https://github.com/aspnet/XRE/pull/1240) to accurately get project information. Right now the application base gives you information about the environment the compile was run from:

A -> B

If you k run in A's directory and you end up compiling B, the IApplicationEnvironment will be for A instead of B.